src/Mod/Part/Gui/ViewProviderBoolean.cpp

```diff
@@ -186,8 +186,8 @@ void ViewProviderMultiFuse::updateData(const App::Property* prop)
         TopTools_IndexedMapOfShape boolMap;
         TopExp::MapShapes(boolShape, TopAbs_FACE, boolMap);
 
-        std::vector<App::Color> colBool;
-        colBool.resize(boolMap.Extent(), this->ShapeAppearance.getDiffuseColor());
+        std::vector<App::Material> colBool;
+        colBool.resize(boolMap.Extent(), this->ShapeAppearance[0]);

```

Comment:
No need to check size first here?

Reply:
No. The PropertyMaterialList code ensures there's always at least one entry